### PR TITLE
Changes overlay lighting vis contents objs -> underlays

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Previously what it would do is create up to three objects; a cone, a lighting image and a holder 
Then whenever the lighting image was placed into the holders vis contents and the holder would be teleported several times when the parent would to turn or move, and the cone would be spun when the parent spun

Instead of doing that it now just translates a single overlay every time that something is updated

## Why It's Good For The Game

Better code

## Changelog
:cl:
refactor: Directional lights might seem more smooth now
/:cl:

### CODE REVIEW

File: code/datums/components/overlay_lighting.dm [hunk 1]
```diff
@@ -429,12 +429,34 @@
 	if(final_distance > SHORT_CAST && !(ALL_CARDINALS & current_direction))
 		final_distance -= 1
 	var/turf/scanning = get_turf(current_holder)
-	for(var/i in 1 to final_distance)
+	for(var/i=1 to final_distance)
```
Comment (Rohesie) [inline code/datums/components/overlay_lighting.dm:432]:
```suggestion
	for(var/i in 1 to final_distance)
```
Is there need for this change?

File: code/datums/components/overlay_lighting.dm [hunk 2]
```diff
@@ -429,12 +429,34 @@
...
+	var/translate_y = translate_x
+	switch(current_direction)
+		if(NORTH)
+			translate_y += 32*final_distance
+		if(SOUTH)
+			translate_y += -32*final_distance
+		if(EAST)
+			translate_x += 32*final_distance
+		if(WEST)
+			translate_x += -32*final_distance
```
Comment (Rohesie) [inline code/datums/components/overlay_lighting.dm:451]:
```suggestion
			translate_y += 32 * final_distance
		if(SOUTH)
			translate_y += -32 * final_distance
		if(EAST)
			translate_x += 32 * final_distance
		if(WEST)
			translate_x += -32 * final_distance
```
Readability.

**Comment by Rohesie** [comment]
Won't changing the `visible_mask`'s `icon`, `alpha` or `color` require it to be recreated and re-applied, since it's an `/image` now?

**Comment by TiviPlus** [comment]
> Won't changing the `visible_mask`'s `icon`, `alpha` or `color` require it to be recreated and re-applied, since it's an `/image` now?

It doesnt need to be recreated just reapplied, it does so now

File: code/datums/components/overlay_lighting.dm (hunk 2 above)
Comment (TiviPlus) [reply code/datums/components/overlay_lighting.dm]: Done

File: code/datums/components/overlay_lighting.dm (hunk 1 above)
Comment (TiviPlus) [reply code/datums/components/overlay_lighting.dm]: Nope, fixed

**Comment by Rohesie** [comment]
Looks good. Is this properly tested locally?

**Comment by TiviPlus** [comment]
Prabably not, it worked pretty okay in testing and any bugs should be minor

**Pull request merged**